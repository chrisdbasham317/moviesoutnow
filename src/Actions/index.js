export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (msg) => ({
  type: 'HAS_ERRORED',
  msg
})

export const setRecentMovies = (movies) => ({
  type: 'SET_RECENT_MOVIES',
  movies
})

export const setGenres = (genres) => ({
  type: 'SET_GENRES',
  genres
})

export const currentUser = currentUser => ({
  type: 'CURRENT_USER',
  currentUser
})

export const toggleModal = toggleState => ({
  type: 'TOGGLE_MODAL',
  toggleState
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const setFavorites = (favorites) => ({
  type: 'SET_FAVORTIES',
  favorites
})