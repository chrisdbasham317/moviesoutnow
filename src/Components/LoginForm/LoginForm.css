.header--login {
  display: flex;
  align-items: center;
}

.header--login__hidden {
  display: none;
}

.header--userinput__div {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-content: flex-start;
  align-items: flex-start;
}

.login__label {
  color: #FFF;
}
.header--userinput__input {
  border-radius: 5px;
  border: none;
  height: 30px;
  width: 180px;
  margin-top: 8px;
  padding-left: 5px;
  font-size: 15px;
}

.header--userinput__div {
  margin-right: 20px;
}

.login--bottomtext {
  color: #5c5a5a;
  margin: 0;
  margin-top: 8px;
}

.login__button {
  border: none;
  height: 30px;
  width: 80px;
  border-radius: 5px;
  background-color: #F541FF;
  color: #FFF;
  font-weight: 700;
  font-size: 15px;
  cursor: pointer;
}

.login__button:disabled {
  background-color: #f8acfc;
  cursor: default;
}
