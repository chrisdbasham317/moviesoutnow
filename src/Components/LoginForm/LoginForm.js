import React from "react"
import "./LoginForm.css"
import movie_time_logo from '../../images/movie_time.png'


export class LoginForm extends React.Component {
  constructor() {
    super()
    this.state = {
      email: "",
      password: ""
    }
  }

  handleChnage = event => {
    this.setState( {[event.target.name]: event.target.value})
  };

  canBeSubmitted() {
   const { email, password } = this.state;
   return email.length > 0 && password.length > 0;
 }

  render() {
    const isEnabled = this.canBeSubmitted();
    return(
      <form className="header--login">
        <div className="header--userinput__div">
          <label
            className="login__label"
            htmlFor="email">Email</label>
          <input
            className="header--userinput__input"
            id="email"
            type="text"
            name="email"
            placeholder="Email"
            onChange={this.handleChnage}
            value={this.state.email} />
          <p className="login--bottomtext" >Create new account</p>
        </div>
        <div className="header--userinput__div">
          <label
            className="login__label"
            htmlFor="password">Passowrd</label>
          <input
            className="header--userinput__input"
            id="password"
            type="password"
            name="password"
            placeholder="Password"
            onChange={this.handleChnage}
            value={this.state.password} />
          <p className="login--bottomtext" >Forgot account?</p>
        </div>
        <button
          type="button"
          className="login__button"
          disabled={!isEnabled}>Login</button>
      </form>
    )
  }
}

export default LoginForm
