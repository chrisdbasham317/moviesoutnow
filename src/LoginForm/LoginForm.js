import React from "react"
import "./LoginForm.css"
import movie_time_logo from '../images/movie_time.png'


export class LoginForm extends React.Component {
  constructor() {
    super()
    this.state = {
      email: "",
      passowrd: ""
    }
  }

  render() {
    return(
      <form className="header--login">
        <div className="header--userinput__div">
          <label
            htmlFor="email">Email</label>
          <input
            className="header--userinput__input"
            id="email"
            type="text"
            placeholder="Email"></input>
          <p>Create new account</p>
        </div>
        <div className="header--userinput__div">
          <label
            htmlFor="password">Passowrd</label>
          <input
            className="header--userinput__input"
            id="password"
            type="text"
            placeholder="Password"></input>
          <p>Forgot account?</p>
        </div>
        <button
          type="button"
          className="login__button">Login</button>
      </form>
    )
  }
}

export default LoginForm