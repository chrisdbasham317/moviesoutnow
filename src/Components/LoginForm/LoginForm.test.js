import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { toggleModal, postUserLogin, hasErrored } from '../../Actions';
import { LoginForm } from './LoginForm';

describe("LoginForm", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <LoginForm
        currentUser={{ id: 3, name: 'Chris' }}
        
      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})