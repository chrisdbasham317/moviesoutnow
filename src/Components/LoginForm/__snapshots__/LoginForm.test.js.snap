// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the snapshot 1`] = `
<div
  className="errorHolder__div"
>
  <form
    className="header--login__hidden"
  >
    <div
      className="header--userinput__div"
    >
      <label
        className="login__label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="header--userinput__input"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value=""
      />
      <p
        className="login--bottomtext new--account__btn"
        onClick={[Function]}
      >
        Create new account
      </p>
    </div>
    <div
      className="header--userinput__div"
    >
      <label
        className="login__label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="header--userinput__input"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <p
        className="login--bottomtext"
      >
        Forgot account?
      </p>
    </div>
    <button
      className="login__button"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </form>
  <div
    className="error--text__div__hidden"
  >
    <img
      alt="Error Icon"
      className="error__icon"
      src="error_icon.png"
    />
    <p
      className="error--text"
    >
      The email and/or passowrd you've entered is incorrect.
    </p>
  </div>
</div>
`;
